pin bump with better architecture

Gasoonjia opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #15016

This diff bump torch pin to 1004.

also expose cuda version into torch_pin.py for better update if needed;

Differential Revision: [D84280496](https://our.internmc.facebook.com/intern/diff/D84280496/)

Gasoonjia committed:
pin bump with better architecture

This diff bump torch pin to 1008.

also expose cuda version into torch_pin.py for better update if needed;

Differential Revision: [D84280496](https://our.internmc.facebook.com/intern/diff/D84280496/)

[ghstack-poisoned]

JacobSzwejbka reviewed (approved):
Review automatically exported from Phabricator review in Meta.

Gasoonjia committed:
Update on "pin bump with better architecture"


This diff bump torch pin to 1008.

also expose cuda version into torch_pin.py for better update if needed;

Differential Revision: [D84280496](https://our.internmc.facebook.com/intern/diff/D84280496/)

[ghstack-poisoned]

Gasoonjia committed:
Update on "pin bump with better architecture"


This diff bump torch pin to 1008.

also expose cuda version into torch_pin.py for better update if needed;

Differential Revision: [D84280496](https://our.internmc.facebook.com/intern/diff/D84280496/)

[ghstack-poisoned]

Gasoonjia committed:
Update on "pin bump with better architecture"


This diff bump torch pin to 1004.

also expose cuda version into torch_pin.py for better update if needed;

Differential Revision: [D84280496](https://our.internmc.facebook.com/intern/diff/D84280496/)

[ghstack-poisoned]
